src/utils/output.ts

```diff
@@ -443,3 +462,29 @@ export class RetrieveResultFormatter implements Formatter<RetrieveResultJson> {
     return packages;
   }
 }
+
+export class MetadataRetrieveResultFormatter implements Formatter<MetadataRetrieveResultJson> {
+  private zipFilePath: string;
+  private files: FileResponse[];
+  public constructor(
+    private result: RetrieveResult,
+    private opts: { 'target-metadata-dir': string; 'zip-file-name': string; unzip: boolean }
+  ) {
+    this.zipFilePath = path.join(opts['target-metadata-dir'], opts['zip-file-name'] ?? 'unpackaged.zip');
+    this.files = sortFileResponses(asRelativePaths(this.result.getFileResponses() ?? []));
+  }
+
+  public getJson(): MetadataRetrieveResultJson {
+    delete this.result.response.zipFile;
+    return { ...this.result.response, zipFilePath: this.zipFilePath, files: this.files };
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  public async display(): Promise<void> {
+    CliUx.ux.log(retrieveMessages.getMessage('info.WroteZipFile', [this.zipFilePath]));
+    if (this.opts.unzip) {
+      const extractPath = path.join(this.opts['target-metadata-dir'], path.parse(this.opts['zip-file-name']).name);
```

Comment:
on line 473, we set `this.zipFilePath` to the default `unpackaged.zip` but it doesn't get defaulted on `this.opts`.  It's always safer to do that kind of stuff in the constructor (have it set defaults on the options object) to avoid this accidental potential undefined.

If you strongly prefer to not modify the options, you could put a private `zipFileName` and have the constructor set that and refer to it here.

Reply:
There's actually not a need to default it anywhere since the flag does that already: https://github.com/salesforcecli/plugin-deploy-retrieve/blob/6df9c943cb08857456513fda5d34e9069517ae54/src/utils/flags.ts#L149

Reply:
even better